docs/user-guide/rs.rst

```diff
@@ -0,0 +1,232 @@
+
+Rejection Sampling Training
+############
+
+After you have fine-tuned a GPT model using Supervised Fine-Tuning (SFT), and trained a reward model as explained in the preceding section, you can start align the policy using rejection sampling.
+
+During rejection sampling training, we conceptually have 2 models interacting with each other:
+
+#. The Policy Network: This is the model we are training, and it should start from an SFT model.
+#. The Reward Model (RM): This model takes a prompt concatenated with a response as input, and outputs a single scalar value: the reward, which the rejection sampling algorithm will try to maximize.
+
+In the most optimized configuration, Aligner will run the actor and initial policy within the same job and the reward model in a separate job. It will then use cpu offloading to load back the corresponding model when needed.
+   
+The next section discusses how to launch each of these two jobs.
+
+Launching the Reward Model and Critic Server
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+To launch the server:
+
+.. code-block:: bash 
+
+   #!/bin/bash
+   CHECKPOINT_NEMO_FILE="/path/to/trained_rm.nemo"
+   GPFS="/path/to/nemo-aligner-repo"
+
+   RESULTS_DIR="critic_results_dir"
+
+   export PYTHONPATH="${NEMO_RLHF_DIR}:${PYTHONPATH}" \
+   && export HYDRA_FULL_ERROR=1 \
+   && python -u examples/nlp/gpt/serve_reward_model.py \
+      --config-path=${CRITIC_CONFIG_PATH} \
+      --config-name=${CRITIC_CONFIG_NAME} \
+      trainer.num_nodes=1 \
+      trainer.devices=8 \
+      ++model.tensor_model_parallel_size=4 \
+      rm_model_file=${RM_NEMO_FILE} \
+      inference.port=${CRITIC_PORT}
+
+
+The above example launches the reward model server on 8 gpus and 1 node. Please make sure to change ``trainer.devices``, ``trainer.num_nodes`` depending on your model size and scale. Aligner will work on any scale. Also make sure to tune the `trainer.rs.inference_micro_batch_size` argument, this sets how big of a batch the RS actor is allowed to send to the critic per DP rank.
+
+Launching the Initial Policy and RS Actor Training
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+The RS Actor training job contains the master controller that makes the HTTP calls to all servers when needed. To launch the RS Actor and Initial Policy server:
+
+.. code-block:: bash 
+
+   GPFS="/path/to/nemo-aligner-repo"
+   TRAIN_DATA_PATH="/path/to/train_prompts.jsonl"
+   VALID_DATA_PATH="/path/to/test_prompts.jsonl"
+
+   PRETRAINED_ACTOR_NEMO_FILE="/path/to/sft_checkpoint.nemo"
+
+   RESULTS_DIR="actor_results_dir"
+
+   export PYTHONPATH="${NEMO_RLHF_DIR}:${PYTHONPATH}" \
+   && export HYDRA_FULL_ERROR=1 \
+   && python -u examples/nlp/gpt/train_gpt_rs_actor.py \
+      --config-path=${CONF_DIR} \
+      --config-name=${CONFIG_NAME} \
```

Comment:
We should drop these so the example works OOTB

Reply:
Done.